pkg/driver/constants.go

```diff
@@ -49,5 +52,23 @@ const (
 	Pending = "PENDING"
 
 	// common properties
-	descriptionKey = "description"
+	descriptionKey     = "description"
+	destroyOnDeleteKey = "destroyOnDelete"
```

Comment:
This is actually specific to the CSP implementation.  A different CSP might choose to destroy on delete by default.  The nimble one does not currently.  We should probably formalize this in the CSP spec.

Reply:
Agreed, the non-nimble CSP should still be able to destroy with/without this metadata key.

Alternatively, why don't we implement the CSP's Delete API to destroy the volume by default unless explicitly requested by the user via SC to retain the volume by just offline it...